cockatrice/src/client/ui/widgets/general/layout_containers/overlap_widget.cpp

```diff
@@ -0,0 +1,195 @@
+ * the widget's width and height, depending on the layout direction. It then updates
+ * the OverlapLayout with these calculated values to ensure the layout is optimized.
+ */
+void OverlapWidget::adjustMaxColumnsAndRows()
+{
+    if (direction == Qt::Vertical) {
+        // Calculate columns based on width for vertical layout
+        int calculatedColumns = overlap_layout->calculateMaxColumns();
+        maxColumns = calculatedColumns;
+        overlap_layout->setMaxColumns(calculatedColumns);
+
+        // Calculate rows based on total item count and columns
+        int calculatedRows = overlap_layout->calculateRowsForColumns(calculatedColumns);
+        maxRows = calculatedRows;
+        overlap_layout->setMaxRows(calculatedRows);
+        qDebug() << "MaxColumns: " << maxColumns << " MaxRows: " << maxRows;
+    } else {
+        // Calculate rows based on height for horizontal layout
+        int calculatedRows = overlap_layout->calculateMaxRows();
+        maxRows = calculatedRows;
+        overlap_layout->setMaxRows(calculatedRows);
+
+        // Calculate columns based on total item count and rows
+        int calculatedColumns = overlap_layout->calculateColumnsForRows(calculatedRows);
+        maxColumns = calculatedColumns;
+        overlap_layout->setMaxColumns(calculatedColumns);
+        qDebug() << " MaxRows: " << maxRows << "MaxColumns: " << maxColumns;
+    }
+
+    overlap_layout->invalidate();
+    overlap_layout->activate();
+}
+
+/**
+ * @brief Updates the maximum number of overlapping items based on new value.
+ *
+ * This method updates the maximum number of columns or rows for the overlap layout
+ * based on the given new value. It adjusts the layout direction accordingly and
+ * triggers a size adjustment for the widget, ensuring the layout reflects the changes.
+ *
+ * @param newValue The new maximum number of overlapping items allowed in the layout.
+ */
+void OverlapWidget::maxOverlapItemsChanged(int newValue)
+{
+    if (direction == Qt::Horizontal) {
+        maxRows = 0;
+        overlap_layout->setMaxRows(0);
+
+        maxColumns = newValue;
+        overlap_layout->setMaxColumns(newValue);
+    } else {
+        maxRows = newValue;
+        overlap_layout->setMaxRows(newValue);
+
+        maxColumns = 0;
+        overlap_layout->setMaxColumns(0);
+    }
+    this->adjustSize();
+    overlap_layout->invalidate();
+}
+
+/**
+ * @brief Changes the layout direction based on the specified new direction.
+ *
+ * This method modifies the layout direction of the OverlapLayout based on the input
+ * string. It updates the direction and triggers a size adjustment for the widget.
+ * Valid inputs are "Qt::Horizontal" and "Qt::Vertical".
+ *
+ * @param newDirection The new layout direction as a QString.
+ */
+void OverlapWidget::overlapDirectionChanged(QString newDirection)
+{
+    if (newDirection.compare("Qt::Horizontal", Qt::CaseInsensitive) == 0) {
```

Comment:
Why QString?

Reply:
It's the way QComboBox stores it/returns it iirc and that's the primary way we allow this to be changed at the moment, in OverlapControlWidget.